bpython/curtsiesfrontend/repl.py

```diff
@@ -844,6 +846,13 @@ def on_control_d(self):
             self.current_line = (self.current_line[:self.cursor_offset] +
                                  self.current_line[(self.cursor_offset + 1):])
 
+    def on_control_o(self):
+        next_idx = -self.rl_history.index + 1
+        next = self.rl_history.entries[next_idx]
+        self.on_enter()
```

Comment:
FYI `on_enter()` has the side effect of resetting the current `rl_history.index` to 0